client/components/SendZoopDialog.tsx

```diff
@@ -0,0 +1,155 @@
+import React, {
+    useState,
+    useEffect 
+} from "react";
+import {useNavigate} from "react-router-dom";
+import Dialog from "@mui/material/Dialog";
+import DialogTitle from "@mui/material/DialogTitle";
+import DialogContent from "@mui/material/DialogContent";
+import DialogContentText from "@mui/material/DialogContentText"
+import DialogActions from "@mui/material/DialogActions";
+import Button from "@mui/material/Button";
+import TextField from "@mui/material/TextField";
+import Link from "@mui/material/Link";
+import Tooltip from "@mui/material/Tooltip";
+import Fab from "@mui/material/Fab";
+import SendIcon from "@mui/icons-material/Send";
+import { 
+    usePostZoopMutation, 
+    useGetAllUsersQuery,
+    useGetMeQuery
+}  from "../features/api";
+import { useSelector } from "react-redux";
+import { RootState} from "../app/store";
+import type { User } from '../../src/types/custom'
+
+const SendZoopDialog = () => {
+    const [open, setOpen] = useState(false);
+    const [username, setUsername] = useState("");
+    const [content, setContent] = useState("");
+    const [zoopSent, setZoopSent] = useState(false);
+    const [lastZoopSentId, setLastZoopSentId] = useState<number | null>(null);
+
+    // Variable used for authorization
+    const token = useSelector((state: RootState) => state.auth.token);
+
+    const currentUser = useGetMeQuery().data;
+
+    const [postZoop, { isLoading: isPostZoopLoading, isError, data: zoopData }] = usePostZoopMutation();
+    const {data: usersData, isLoading: isGetAllUsersLoading, error } = useGetAllUsersQuery() 
+
+    const navigate = useNavigate();
+
+    const handleSendZoopClick = () => {
+        if (!usersData) {
+            throw new Error('No users to send Zoop to');
```

Comment:
I think this is fine - but you should also ensure that the component doesn't even render until the users list has loaded